Quest Rework.

Kanzaji opened the pull request:
I've got awesome idea what to do, and I decided to fully port quests from E6 to E8, redo quest lines that need update (like Create one), hide ones for mods that aren't a thing in E8 atm.
Blah blah blah -> Get the most of E6 quests in E8.

I have no idea if this is good idea, or if any work has been started on dev side, thats why this is just empty draft. I will start redoing some of the quests atm, but let me know if this is good idea 😄 

PS: Yes this is my way to have a break from CC:T quests in E6, and its to do more quests but on different topics 😆

Kanzaji commented:
And I of course forgot to mark is as a draft 😅

NielsPilgaard commented:
Ehm.. we already have the quests from e6 in e8 though :P

Kanzaji commented:
Like... all? 
If so I'm really outdated somehow with E8 😅

Kanzaji commented:
Okay I did boot up Dev, and still a lot of quests are missing and some are broken.
I should probably write it differently what I want to do: I would like to fix broken quests, re-do ones that aren't revelant to the version of the mod (I remember create being the one), and add those that are missing but mods are in the modpack :D
So:
Quests to add:
- Mekanism
- Blood Magic
- Botania
- Industrial Foregoing
- Nature's aura
- PNC
- Powah -> If it will be added to the modpack, as there is suggestion about it :D
- Thermal Series
- Tinkers Construct

To fix: Anything that seems to be broken / missing some quests

To redo: 
- Create

List can change if I found something needs to be re-done as create :D

Kanzaji commented:
I hope that makes sense now

Kanzaji commented:
Buidling quests done. Removed few of the quests as Astral Sorcery, Mod for projecting your house and few others aren't in E8. Also few quests changed to fit what is actually in the modpack with the same functionailty
![obraz](https://user-images.githubusercontent.com/60540476/184558449-529552c4-f596-4509-a140-051072d4a606.png)
Only problem are loot boxes, as those are a bit broken (for some reason) and fact that for Wand of Symmetry and Extendo Grip there is no recipe made :P

NielsPilgaard commented:
Ah that makes more sense, yes please :P 

How are loot boxes broken?

Kanzaji commented:
👍 And again sorry for explaining that soo bad first time 😅 I was a bit tired.

Loot Boxes sometimes doesn't spawn anything. Those delight ones seems to be working fine, but mod specific ones have that issue

Kanzaji commented:
![obraz](https://user-images.githubusercontent.com/60540476/184592630-550fe00e-7ad5-4e85-9962-f915b128c965.png)

Tools and Armor done!
Only problem is that iron jetpacks confuse hell out of FTB Quests (#213) and again loot boxes.
Also I want to add Morp-o-tool recipe that was in E6, that gives it all wrencehes from various mods :D But thats again, for later
Removed I think only 1 quest, re-done Iron jetpacks (that I will probably delete later on, as ^), and fixed a few.

Kanzaji commented:
Tetra quests Done! I'm not sure about Forge Hammers, as it looks like it isn't implemented in 1.18.
Did try to do everything quests tells you to, and it seems everything works fine :D (Even loot boxes!)

![obraz](https://user-images.githubusercontent.com/60540476/184617188-43f2e589-9445-4342-9402-29174c9b5314.png)

Also trying to work out how to fix iron jetpacks, as NBT Filters doesn't seems to allow for showing of crafting recipes (and I can't use AND for some reason to add wooden jetpack as second one, so you can check out recipe)

Kanzaji commented:
Forgot to move that Tier 6 Hammer a bit higher, so it looks better

![obraz](https://user-images.githubusercontent.com/60540476/184617757-646997ab-2523-47b4-aea6-fa186dc9bd96.png)

mickelus commented:
Not sure if this is a legit attempt at improving the quests, so I apologize if this is the wrong place to provide some input.

The tetra structures won't be available in the 1.18 version, the structure loot is craftable and the hammer progression has been adjusted instead. From what I can remember the enigmatica playerbase did not enjoy the gameplay around the structures, so this could be an opportunity to tweak this into something they could enjoy. The loot is currently craftable using netherite, but that could be made more interesting with other materials or perhaps custom schematics to craft those modules directly in the bench instead.

It also looks like the schematics and treatises are missing from the quests, and it would be help players to discover the mod if the quests delved deeper into the use of the ruins loot (e.g. there are three different uses for the extractors).

mickelus commented:
Also noticed a bunch of typos and grammatical errors in the quest updates, there's lots of tools available to help with that if you're struggling.

Kanzaji commented:
I'm mostly porting quests from E6 and making those that are already ported work ^^" So typos are probably from E6 somehow. I will check my quests for typos later, but not all as most of those are literally the same in E6.

Also I didn't touch much Tetra so thats why there isn't much new, I just yeeted stuff that isn't in E8 and checked if everything works :D If you want you can send me stuff to add for Tetra quest line and I will try to do something with it, but hey it is better than it was!

Also can you tell me in which quests you found typos? I'm curious if those are mine or from E6.

Kanzaji commented:
Also I just realised you are the dev of Tetra 😅 Well yea I would definitely need to learn Tetra a bit more to add info about schematics and other stuff. But thx for letting know about changed hammer progression, definitely need to fix that

Kanzaji commented:
Tinkers Construct are done! I needed to add a tag to Tinkers Anvils as they had any unique ones, so it was impossible to fix one of the quests.

![obraz](https://user-images.githubusercontent.com/60540476/184656036-a48ae810-9df5-4397-bff2-6efa505e8ec5.png)

Known problems:
Of course loot boxes are broken.
Piggy back doesn't have recipe
Recipe for Slime armor isn't showing up in JEI (Recipe works, so probably bug in JEI)

Kanzaji commented:
Did change a bit Tetra quests again with info I've got from mickelus. I will try to do more when I would actually learn the mod better ^^" But no promises.
![obraz](https://user-images.githubusercontent.com/60540476/184670155-2c9006df-b9da-4226-baaf-993b76b26939.png)

NielsPilgaard commented:
Nice progress so far, good work!

Kanzaji commented:
Storage chapter nearly done! (Only thing left is Armory cabinet, that I can't get how it works yet)
Added Storage tags from E6, also added additional ones for Sophisticated Storage:

- forge:storage/basic/{tier}

Also added new sections!:

- Functional Storage
- Sophisticated Storage:
- A bit of AE2

![obraz](https://user-images.githubusercontent.com/60540476/184715564-4bf30aef-ff56-4332-95e4-ca1e82841a9e.png)

Kanzaji commented:
After a lot of struggle, and having fun trying different things, I found working, but wonky solution
I did use hidden quests, a lot of them, and game stages to get those quests working, but hey its less wonky than I would use NBT Filter giving out an error ^^"

Kanzaji commented:
Functional Storage changed a tiny bit, as I did get new info about it after last ss was made ^^" Anyway Storage is done!
![obraz](https://user-images.githubusercontent.com/60540476/184732505-8bb32f5f-2107-4d4e-b64b-fbb1f89f6982.png)

Kanzaji commented:
I did lie, I forget to add few things to Sophisticated Storage 😅 Anyway fixed!

Kanzaji commented:
Oh. JEI in DEV is broken...

![obraz](https://user-images.githubusercontent.com/60540476/184828242-92a19fe4-f38e-4711-ab26-ed737109822d.png)

I don't have idea how to "update" mods here, but I found out that JEI 232 works fine at least with Tinkers and Create

![obraz](https://user-images.githubusercontent.com/60540476/184828337-5e7043db-a6ed-4308-9006-be03293fc20b.png)

I will do Issue later on about it, but for now back to doing quests!

Kanzaji commented:
OH, so thats why I thought there are no recipe for Wand of Symmestry and the rest. JEI was broken and those are in mechanical crafter crafted 😅
![obraz](https://user-images.githubusercontent.com/60540476/184828760-cd300eb4-396a-4298-a445-650bf81c5698.png)

Kanzaji commented:
Automation chapter done! There were only few fixes that needed to be made, and pretty pipes section to delete ^^" But otherwise it should be fine :D

![obraz](https://user-images.githubusercontent.com/60540476/184849691-adfdfb82-6a8c-48d5-8493-e322e5e27262.png)

I know there is one quest with missing stuff, but thats entangled block and this mod is ported to 1.18.2, so it would be nice to have it in E8 (If suggestion #218 will not be accepted, well I will delete it)

Kanzaji commented:
Refined Storage was already on E8, but I did add few missing things that (Tbh I have no idea why) were deleted from E8, also new quest about Crafter Chaining as Iron crafters aren't a thing in E8

![obraz](https://user-images.githubusercontent.com/60540476/184853116-77c20324-7390-4359-bd2d-a70a19ee66fd.png)

Kanzaji commented:
Integrated Dynamics is another one that just... works after port from E6.
Fixed few things that were even broken in E6, otherwise added few missing quests that were missing from E8

![obraz](https://user-images.githubusercontent.com/60540476/184855521-fc991ed6-a00b-4e9f-8e5e-cab9edb51ee6.png)

Kanzaji commented:
Oops, added one file that shouldn't be there 😅

Kanzaji commented:
IE quests fixed! I did add a few machines that werent in the quest originally, and also added a note about Immersive Petroleum not being ported yet, as I hope it will be back.
Loot boxes seems to work :D Another one out of the way

![obraz](https://user-images.githubusercontent.com/60540476/184900202-d61876e8-ffd3-481d-ad12-135c3fb544f1.png)

Kanzaji commented:
Industrial Foregoing fixed. Added a lot of missing stuff that wasnt even in E6 (Also created PR adding those changes to E6 :D)
Nothing much else changed, everything works except loot boxes.

![obraz](https://user-images.githubusercontent.com/60540476/184909175-638b6c5a-9657-4ace-8d00-ff9a3f30611b.png)
![obraz](https://user-images.githubusercontent.com/60540476/184915694-20071c42-ebc1-415d-af55-d414bd1fa6ce.png)

Kanzaji commented:
Mekansim quests done! Not much changed, only few fixes and ore processing rework
In terms of amount of quests - no changes
Also loot boxes doesn't wok as always.

![obraz](https://user-images.githubusercontent.com/60540476/184932614-69feada2-89d1-47da-8002-97beeea8724e.png)

Kanzaji commented on config/ftbquests/quests/chapters/natures_aura.snbt:
```diff
@@ -99,14 +108,21 @@
+				{
+					id: "14A7BCABC52C6E6D"
+					type: "item"
+					item: "farmingforblockheads:market"
```
Didn't touch natures aura yet ^^" This one is still raw from E6.

NielsPilgaard replied:
Oh, okay then.

Kanzaji commented on kubejs/server_scripts/base/tags/items/forge/storage/sophisticated_storage.js:
```diff
@@ -0,0 +1,10 @@
+onEvent('item.tags', (event) => {
+    event.add('forge:storage/basic/wooden','#sophisticatedstorage:base_tier_wooden_storage','sophisticatedstorage:basic_tier_upgrade','sophisticatedstorage:shulker_box')
```
I did that mostly for quests about sophisticated storage, but to not make those only "for quests purposes" I tried making them actually somewhat useful.
If you want I can change those to `sophisticatedstorage:` and change a bit naming, in result making those tags tier own group.

NielsPilgaard replied:
I think that'd be better, yes please.

Kanzaji replied:
Alrighty so script is done, tag names changed to storage_{tier}
I would normally use mod name, but it would be a pain to write this entire mod name everytime you want to search some iron tiered chests 😅 If you have better idea for a tag name, I'm open for suggestion

MuteTiefling commented on config/ftbquests/quests/chapters/mekanism.snbt:
```diff
@@ -353,7 +350,7 @@
 				"Alloys will be needed commonly for further progression and without a little extra effort, they can become quite costly to make. Enter the Enrichment Chamber. "
 				""
-				"Running Ores through the Enrichment Chamber will double the yield from each ore. "
+				"Running Raw Ores through the Enrichment Chamber will increase the yield from each ore to 1.33. Ore doubling is still a thing here, but you need to process full ore blocks for that."
```
`still a thing here` suggests the player knows what came before. Making relative statements to prior versions isn't really helpful as many won't have that knowledge. So I'd just avoid it outright. 

`Running Raw Ores through the Enrichment Chamber will increase the yield by 33 percent, while running Full Ore Blocks will double the output`

MuteTiefling commented on config/ftbquests/quests/chapters/mekanism.snbt:
```diff
@@ -471,8 +468,8 @@
-					title: "Orange"
-					item: "simplefarming:orange"
+					title: "Melon"
+					item: "minecraft:melon_slice"
```
No oranges in e8? Sad orange crush joke noises.

Just drop the melon entirely. The joke doesn't work :D

Kanzaji replied:
Ah thats why it was here hah. I don't know those jokes unfortunetaly😅
How about we will add Enigmatica™️ Orange :D?

Kanzaji replied:
(Nah I did delete melon completely, if some mod later on will add orange, I will re-add it here :D)

MuteTiefling commented on config/ftbquests/quests/chapters/mekanism.snbt:
```diff
@@ -676,19 +674,22 @@
 			subtitle: "Not Enough Minerals"
 			description: [
-				"With enough effort, Mekanism allows for the extraction of up to five ingots per ore, but this is a very involved process and will require serious effort to set up. "
+				"With enough effort, Mekanism allows for the extraction of up to 3 (3.33 exactly) ingots per ore, but this is a very involved process and will require serious effort to set up."
```
This was talking about the absolute cap on mek processing, if that's still 5x, it should still mention it. Here would be a good place to really spell out that Raw Ore and Ore Blocks have different rates with Raw Ore being less and just leave it at that. Let them discover the exact rates through JEI.

MuteTiefling commented on config/ftbquests/quests/chapters/mekanism.snbt:
```diff
@@ -676,19 +674,22 @@
+				"With enough effort, Mekanism allows for the extraction of up to 3 (3.33 exactly) ingots per ore, but this is a very involved process and will require serious effort to set up."
 				""
-				"However, between doubling and quintupling, it can also triple or quadruple yields from ores. Tripling is very attainable early on and well worth setting up."
+				"However, between increasing yield to 1.33 and 3.33, it can also increase it to 2 or 2.66. Doubling with fortune pick is very attainable early on and well worth setting up."
```
Wouldn't even call out the worse rates here. If it can still do 3x and 4x, just tell them. Also not the place to mention fortune mining as that's unrelated to Mek processing.

MuteTiefling commented on config/ftbquests/quests/chapters/mekanism.snbt:
```diff
@@ -1393,7 +1394,7 @@
 			x: 7.5d
 			y: 9.0d
-			description: ["Oh! You're still here?"]
+			description: ["Oh! You're still here? Have some cookies for your determination!"]
```
The 'have a cookie' was implied by the reward ^_~ No need to state it.

Kanzaji replied:
Right :D

MuteTiefling commented on config/ftbquests/quests/chapters/mekanism.snbt:
```diff
@@ -2520,16 +2520,18 @@
 				"The ultimate in ore processing efficiency… or is it? "
 				""
-				"Mekanism’s highest form of ore processing ends at quintupling and requires a hefty amount of infrastructure to access. As with quadrupling, the exact process can be determined from JEI by looking at metal crystals. "
+				"Mekanism’s highest form of ore processing ends here, at 3.33 ingot per raw ore (or 5 per ore block), and requires a hefty amount of infrastructure to access. As with last tier, the exact process can be determined from JEI by looking at metal crystals. Invest in some sort of auto-breaker of ores with Fortune, as this could boost per ore block yield way above 5 ingots!"
```
As above, don't even mention the worse rates. We're talking about max output, not the full spectrum.

NielsPilgaard commented on kubejs/server_scripts/base/tags/items/tconstruct/tinkers_anvil.js:
```diff
@@ -1,3 +1,3 @@
 onEvent('item.tags', (event) => {
-    event.add('tconstruct:tinkers_anvils', ['tconstruct:scorched_anvil', 'tconstruct:tinkers_anvil']);
```
remember to update file placement too please :D

Kanzaji replied:
Ah, sure thing!
(Thank god everything comes out as I go, because fixing everything again after you tried to fix it is annoying 😆 )

Kanzaji commented:
Okay I think that's everything for now 😅 I can come back to fixing another quests!

NielsPilgaard reviewed (approved):
Nice work! Thank you :D
